### PR TITLE
feat: add endpoint to find a user by username

### PR DESCRIPTION
**Pull request by brandonroberts**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR:
- Adds the `/users/:username` endpoint to find a user by username

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/user/user.service.ts [hunk 1]
```diff
@@ -39,6 +39,27 @@ export class UserService {
...
 
+  async findOneByUsername (username: string): Promise<DbUser> {
+    const queryBuilder = this.baseQueryBuilder();
+
+    queryBuilder.where("LOWER(login) = LOWER(:username)", { username });
+
+    let item: DbUser | null;
+
+    try {
+      item = await queryBuilder.getOne();
```
Comment (0-vortex) [inline src/user/user.service.ts:50]: Are you sure you need the try/catch here? 😅 

You're not really doing anything with the error, this is like hiding it from our system

**Review by 0-vortex — CHANGES_REQUESTED** [review]
Not sure about empty errors, would be best to just leave the default behavior 👍

File: src/user/user.service.ts (hunk 1 above)
Comment (brandonroberts) [reply src/user/user.service.ts]: ✂️

File: src/user/user.service.ts [hunk 2]
```diff
@@ -39,6 +39,20 @@ export class UserService {
     return item;
   }
 
+  async findOneByUsername (username: string): Promise<DbUser> {
+    const queryBuilder = this.baseQueryBuilder();
+
+    queryBuilder.where("LOWER(login) = LOWER(:username)", { username });
+
+    const item: DbUser | null = await queryBuilder.getOne();
```
Comment (0-vortex) [inline src/user/user.service.ts:47]: `getOneOrFail` maybe?

Comment (brandonroberts) [reply src/user/user.service.ts:47]: We aren't using this anywhere else, will the error still be presented the same way?

Comment (brandonroberts) [reply src/user/user.service.ts:47]: This throws a 500, where we need a try/catch

Comment (0-vortex) [reply src/user/user.service.ts:47]: ok with throwing a new one on top of https://docs.nestjs.com/exception-filters#built-in-http-exceptions

Comment (brandonroberts) [reply src/user/user.service.ts:47]: So catch and rethrow `NotFoundException`?

Comment (0-vortex) [reply src/user/user.service.ts:47]: no, like the first time you did, when you threw on null and let all the other errors propagate before that

Comment (0-vortex) [reply src/user/user.service.ts:47]: or well, second :D

**Pull request merged**